### ISSUE TITLE
[Steam Remote Play] Resolutions above 2560x1600 cause a still image, often a black screen, to be streamed in proton games

### ISSUE THREAD

**Issue by ryao** [ValveSoftware/steam-for-linux#8554]
#### Your system information

* Steam client version (build number or date): Apr 26 2022, at 04:19:41
* Distribution (e.g. Ubuntu): Gentoo
* Opted into Steam client beta?: [Yes/No] Yes
* Have you checked for system updates?: [Yes/No] Yes

#### Please describe your issue in as much detail as possible:
I have a 4K display. If I use the steamlink client on an Android TV or iPad to try to play games remotely, I get a black screen on the remote display. If I change the host screen resolution to 1080p via KDE settings, nvidia-settings or xrandr, it works. If I try using a native Linux game, such as Plague Inc: Evolved, it also works.

This is also an issue for people with ultrawide monitors:

https://www.reddit.com/r/linux_gaming/comments/uccw7n/comment/i6ec4of/

#### Steps for reproducing this issue:

1. Use a 4K display.
2. Setup steam link.
3. Connect to steam link.
4. Try to play *any* Proton game. My suggestion is `Destroy All Humans` since I use it the most to test.

#5770 is already open for a black screen issue, but it is not clear if the root cause is the same, so I am opening a new issue for this.

**Comment by ryao** [comment]
This is more of an issue for Linux users than the steam hardware survey would initially suggest. When looking at all operating systems combined, 67.24% of users use 1080p displays, but when looking at it for Linux users, only 53.29% use 1080p displays:

https://store.steampowered.com/hwsurvey/?platform=linux

That means that as much as half of all Linux users are potentially affected by this. I imagine many of them try Steam Remote Play, see that it is broken and just do not bother with it again, instead of filing bug reports.

**Comment by ryao** [comment]
Setting "Change desktop resolution to match streaming client" made it work, when in the past, it would cause issues and left me with a black screen. It might be working because I used this unofficial NvFBC patch:

https://github.com/keylase/nvidia-patch

**Comment by ryao** [comment]
More testing with the following settings:

* 4K host screen
* Change desktop resolution to match streaming client
* Use NVFBC capture on Nvidia GPU (plus the unofficial patch)
* Destroy All Humans in Windowed Mode

Now I can cycle through these resolutions without anything going wrong:

* 1600x900
* 1680x1050
* 1920x800
* 1920x1080
* 1920x1200
* 1920x1280
* 2560x1080
* 2560x1440
* 2560x1600

The moment I try any of these resolutions, the stream shows only the last frame:

* 2880x1620
* 3200x1800
* 3840x1600
* 3840x2160

I can alt tab to the desktop on the host and the desktop is displayed, and if I alt-tab back and set one of the working resolutions, things work. The problem is not that the game's resolution has exceeded the iPad's, since the ipad has a 2224x1668 native resolution and 2560x1600 worked fine, yet 2880x1620 broke. It is not exceeding the iPad's vertical resolution either since 2880x1620 is broken.

Also, if I switch to windowed fullscreen or fullscreen in the game, things break, much like they do on windowed 3840x2160.

I have a 4K Android TV where I could test, but since it is on an entirely different floor, testing different things at the host to see what happens on it is not as easy as doing the same with my iPad. In any case, the reason I was using the ipad at all was because this was broken on the Android TV and I did not want to keep running up and down stairs to test what happens with different settings. :/

**Comment by ryao** [comment]
I also turned gsync off on my machine and it made no difference, so this is not related to VRR.

I also just turned off `Change desktop resolution to match streaming client` to check for the placebo effect and unfortunately, things seemed to still work as I previously described. I am not sure what that setting does, but the game remembered the previous windowed mode settings and streamed to my ipad without a problem. :/
